BASE/Test/ServerTelemetryChannel.Test/TelemetryChannel.Tests/Implementation/TelemetryBufferTest.cs

```diff
@@ -379,6 +380,145 @@ public void DoesNotContinueOnCapturedSynchronizationContextToImprovePerformance(
             }
         }
 
+        [TestClass]
+        public class ManualFlushAsync : TelemetryBufferTest
+        {
+            [TestMethod]
+            [Timeout(10000)]
+            public void DoesntSerializeTelemetryIfBufferIsEmpty()
+            {
+                bool telemetrySerialized = false;
+                var serializer = new StubTelemetrySerializer();
+                serializer.OnSerialize = telemetry =>
+                {
+                    telemetrySerialized = true;
+                };
+                var telemetryBuffer = new TelemetryBuffer(serializer, new StubApplicationLifecycle());
+
+                telemetryBuffer.FlushAsync(CancellationToken.None).GetAwaiter().GetResult();
```

Comment:
:bulb: You can use `await` in test methods. Just make the signature `public async Task DoesntSerialize...`

Reply:
:bulb: Also check the result of the FlushAsync task. It should be 'true', right?